File: csharp/DesiArchitect.DesignPatterns/05_Singleton/After/AppConfig.cs
namespace DesiArchitect.DesignPatterns.Singleton.After;

public class AppConfig : IAppConfig
{
    public string ApiKey => "sk-live-demo-key-12345";
    public string DatabaseConnection => "Server=prod-db;Database=Orders;";

    public AppConfig()
    {
        Console.WriteLine("  [After] AppConfig created by DI container (singleton lifetime)");
    }
}

File: typescript/src/05-singleton/after/app-config.ts
import type { IAppConfig } from './i-app-config.js';

export class AppConfig implements IAppConfig {
  readonly apiKey = 'sk-live-demo-key-12345';
  readonly databaseConnection = 'Server=prod-db;Database=Orders;';

  constructor() {
    console.log('  [After] AppConfig created by container (singleton lifetime)');
  }
}